02-peer-protocol.md

```diff
@@ -607,13 +619,22 @@ A receiving node which closes an incoming HTLC in response to an
 `update-fail-htlc` message on an offered HTLC MUST copy the `reason`
 field to the outgoing `update-fail-htlc`.
 
+A receiving node MUST fail the channel if the `BADONION` bit in
+`failure-code` is not set for `update-fail-malformed-htlc`.
```

Comment:
Only your direct peer can send you update-fail-malformed-htlc: if it receives it from an upstream node it converts it to a normal update-fail-htlc.

It should *never* be trying to induce you send an update-fail-htlc with some random failure code!